src/qaqc/download_and_flag_fxns/generate_site_metaparam.R

```diff
@@ -0,0 +1,36 @@
+#' @title Generate site metaparameter data from the HydroVu API.
+#'
+#' @description
+#' A function that generates a site metaparameter dataframe from the HydroVu API.
+#' A metaparameter is a parameter that is used to generate flags for parameters
+#' other than itself.
+#'
+#' @param api_data A dataframe with the munged API data.
+#' 
+#' @param require An upstream dependency (i.e., alternative to `{necessary object}`).
+#'
+#' @return A dataframe with site metaparameter data that will be joined to the other
+#' site-parameter data.
+#'
+#' @examples
+#' generate_site_metaparam(site_arg = "archery", metaparameter_arg = "Battery Level", api_data = incoming_data_collated_csvs)
+#' 
+#' @seealso [summarize_site_param()]
+
+generate_site_metaparam <- function(api_data, require = NULL) {
+
+    sites <- unique(api_data$site)
+    metaparameters <- c("Temperature", "Battery Level", "Baro", "External Voltage")
```

Comment:
I think I know why some sites don't have metadata - it is because they are AT 500s, so those data are not included in the TROLLs themselves. If we end up requiring this information (aka, not a priority right now) we will need to write some code for pulling additional data from the VuLinks, which are excluded from this QAQC workflow as it is currently.

Reply:
That makes sense!